### PR TITLE
Move backoff from util/backoff to dskit/backoff. Rename backoff config to config for clarity

### PR DESCRIPTION
**Pull request by treid314**
Signed-off-by: Tyler Reid <tyler.reid@grafana.com>

### CODE REVIEW

File: backoff/backoff.go [hunk 1]
```diff
@@ -8,31 +8,31 @@ import (
 	"time"
 )
 
-// BackoffConfig configures a Backoff
-type BackoffConfig struct {
+// Config configures a Backoff
+type Config struct {
```
Comment (aknuds1) [inline backoff/backoff.go:12]: Nice change.

**Review by aknuds1 — APPROVED** [review]
Nice one!

File: backoff/backoff.go [hunk 2]
```diff
@@ -8,31 +8,31 @@ import (
...
 	MinBackoff time.Duration `yaml:"min_period"`  // start backoff at this level
 	MaxBackoff time.Duration `yaml:"max_period"`  // increase exponentially to this level
 	MaxRetries int           `yaml:"max_retries"` // give up after this many; zero means infinite retries
 }
 
 // RegisterFlags for BackoffConfig.
```
Comment (simonswine) [inline backoff/backoff.go:18]:
```suggestion
// RegisterFlags for Config.
```

File: backoff/backoff.go [hunk 3]
```diff
@@ -8,31 +8,31 @@ import (
...
+	cfg          Config
 	ctx          context.Context
 	numRetries   int
 	nextDelayMin time.Duration
 	nextDelayMax time.Duration
 }
 
 // NewBackoff creates a Backoff object. Pass a Context that can also terminate the operation.
-func NewBackoff(ctx context.Context, cfg BackoffConfig) *Backoff {
+func NewBackoff(ctx context.Context, cfg Config) *Backoff {
```
Comment (simonswine) [inline backoff/backoff.go:35]:
```suggestion
func New(ctx context.Context, cfg Config) *Backoff {
```

File: backoff/backoff_test.go [hunk 4]
```diff
@@ -85,7 +85,7 @@ func TestBackoff_NextDelay(t *testing.T) {
 		t.Run(testName, func(t *testing.T) {
 			t.Parallel()
 
-			b := NewBackoff(context.Background(), BackoffConfig{
+			b := NewBackoff(context.Background(), Config{
```
Comment (simonswine) [inline backoff/backoff_test.go:88]:
```suggestion
			b := New(context.Background(), Config{
```

**Review by simonswine — APPROVED** [review]
LGTM, also see a couple of nits. No hard feelings, so feel free to ignore :slightly_smiling_face:

File: backoff/backoff.go [hunk 5]
```diff
@@ -8,31 +8,31 @@ import (
...
+type Config struct {
 	MinBackoff time.Duration `yaml:"min_period"`  // start backoff at this level
 	MaxBackoff time.Duration `yaml:"max_period"`  // increase exponentially to this level
 	MaxRetries int           `yaml:"max_retries"` // give up after this many; zero means infinite retries
 }
 
-// RegisterFlags for BackoffConfig.
-func (cfg *BackoffConfig) RegisterFlags(prefix string, f *flag.FlagSet) {
+// RegisterFlags for Config.
+func (cfg *Config) RegisterFlags(prefix string, f *flag.FlagSet) {
```
Comment (pracucci) [inline backoff/backoff.go:19]: [nit] By convention, we're used to call `RegisterFlagsWithPrefix()` the function to register flags with prefix. I would adhere here too.

**Review by pracucci — APPROVED** [review]
LGTM! Just a nit.

**Pull request merged**